src/test/java/de/siegmar/logbackgelf/pool/SimpleObjectPoolTest.java

```diff
@@ -122,19 +125,13 @@ void infinitePollWaitTime() throws InterruptedException {
                 new SimpleObjectPool<>(factory, 1, -1, 100, 0);
 
         final MyPooledObject o1 = pool.borrowObject();
-        final long before = System.currentTimeMillis();
+        final long before = System.nanoTime();
 
-        new Thread(() -> {
-            try {
-                Thread.sleep(200);
-            } catch (InterruptedException e) {
-                throw new RuntimeException(e);
-            }
-            pool.returnObject(o1);
-        }).start();
+        Executors.newScheduledThreadPool(1)
+                .schedule(() -> pool.returnObject(o1), 200, TimeUnit.MILLISECONDS);
 
         final MyPooledObject o2 = pool.borrowObject();
-        final long after = System.currentTimeMillis();
+        final long after = System.nanoTime();
 
         assertNotNull(o2);
         assertTrue(after - before > 190);
```

Comment:
This also needs to be adjusted for nanoTime

Reply:
Fixed